test/cttestsrv/sth.go

```diff
@@ -0,0 +1,35 @@
+package cttestsrv
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"crypto/sha256"
+
+	"github.com/google/certificate-transparency-go"
+	cttls "github.com/google/certificate-transparency-go/tls"
+)
+
+func signSTH(signer *ecdsa.PrivateKey, sth *ct.SignedTreeHead) error {
+	sthBytes, err := ct.SerializeSTHSignatureInput(*sth)
+	if err != nil {
+		return err
+	}
+
+	h := sha256.New()
```

Comment:
> even inline it into the args of Sign

Unfortunately putting it inline doesn't work. `Sign` takes `[]byte` but `sha256.Sum256` returns `[32]byte`. You can't put `sha256.Sum256(foo)[:]` inline because that tries to take a slice of an unaddressable value. It needs to be an intermediate var that gets used as `var[:]` in the args.